README.md

```diff
@@ -1459,6 +1460,24 @@ def finish_guided_session
   [foreman](https://github.com/ddollar/foreman) to manage them.
 <sup>[[link](#foreman)]</sup>
 
+## Auto loader rules
+
+Prefer nesting a class within a module instead of using module as part of class name:
+
+  ```ruby
+  module User
+    class Admin
+    end
+  end
+
+  class User::Admin
+  end 
+  ```
+
+If a class is nested within a module you do not have to provide the fully qualified class name.
+In order to avoid naming conflicts do not autoload such classes. Require the class with
+`require_dependency` where it is needed.
```

Comment:
> Require the class with`require_dependency` where it is needed.

'where it is needed' does not explain (wait for it) ... where it is needed 😀  In which situation is itnecessaryy to use require_dependency?